### PR TITLE
[Merged by Bors] - feat: port Topology.Hom.Open

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Hom/Open.lean [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    obtain ⟨⟨_, _⟩, _⟩ := g
+    congr
+  map_continuous f := f.continuous_toFun
+  map_open f := f.map_open'
+
+@[simp]
+theorem toFun_eq_coe {f : α →CO β} : f.toFun = (f : α → β) :=
+  rfl
+#align continuous_open_map.to_fun_eq_coe ContinuousOpenMap.toFun_eq_coe
+
```
Comment (urkud) [inline Mathlib/Topology/Hom/Open.lean:82]:
```suggestion
theorem toFun_eq_coe {f : α →CO β} : f.toFun = (f : α → β) :=
  rfl
#align continuous_open_map.to_fun_eq_coe ContinuousOpenMap.toFun_eq_coe

@[simp]
theorem coe_toContinuousMap (f : α →CO β) : (f.toContinuousMap : α → β) = f := rfl
```

File: Mathlib/Topology/Hom/Open.lean [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  map_continuous f := f.continuous_toFun
+  map_open f := f.map_open'
+
+theorem toFun_eq_coe {f : α →CO β} : f.toFun = (f : α → β) :=
+  rfl
+#align continuous_open_map.to_fun_eq_coe ContinuousOpenMap.toFun_eq_coe
+
+@[simp]
+theorem coe_toContinuousMap (f : α →CO β) : (f.toContinuousMap : α → β) = f := rfl
+@[ext]
```
Comment (urkud) [inline Mathlib/Topology/Hom/Open.lean:84]:
```suggestion
theorem coe_toContinuousMap (f : α →CO β) : (f.toContinuousMap : α → β) = f := rfl

@[ext]
```

File: Mathlib/Topology/Hom/Open.lean [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+variable {F α β γ δ : Type _}
+
+/-- The type of continuous open maps from `α` to `β`, aka Priestley homomorphisms. -/
+structure ContinuousOpenMap (α β : Type _) [TopologicalSpace α] [TopologicalSpace β] extends
+  ContinuousMap α β where
+  map_open' : IsOpenMap toFun
+#align continuous_open_map ContinuousOpenMap
+
+-- mathport name: «expr →CO »
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Hom/Open.lean:41]:
```suggestion
```

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Topology/Hom/Open.lean [hunk 4]
```diff
@@ -76,22 +74,19 @@ instance : ContinuousOpenMapClass (α →CO β) α β
...
-instance : CoeFun (α →CO β) fun _ => α → β :=
-  FunLike.hasCoeToFun
-
-@[simp]
 theorem toFun_eq_coe {f : α →CO β} : f.toFun = (f : α → β) :=
   rfl
 #align continuous_open_map.to_fun_eq_coe ContinuousOpenMap.toFun_eq_coe
 
+@[simp]
+theorem coe_toContinuousMap (f : α →CO β) : (f.toContinuousMap : α → β) = f := rfl
```
Comment (eric-wieser) [inline Mathlib/Topology/Hom/Open.lean:82]: Maybe `--porting note: new`?

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Topology/Hom/Open.lean (hunk 4 above)
Comment (urkud) [reply Mathlib/Topology/Hom/Open.lean:82]:
```suggestion
@[simp] -- porting note: new, simpNF of `toFun_eq_coe`
theorem coe_toContinuousMap (f : α →CO β) : (f.toContinuousMap : α → β) = f := rfl
```

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**